server/handler/auth.go

```diff
@@ -6,48 +6,55 @@ import (
 
 	"github.com/sirupsen/logrus"
 	"github.com/src-d/code-annotation/server/repository"
+	"github.com/src-d/code-annotation/server/serializer"
 	"github.com/src-d/code-annotation/server/service"
 )
 
 // Login handler redirects user to oauth provider
-func Login(oAuth *service.OAuth) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
+func Login(oAuth *service.OAuth) RequestProcessFunc {
+	return func(r *http.Request) (resp *serializer.Response, errs []*serializer.Error) {
 		url := oAuth.MakeAuthURL()
-		http.Redirect(w, r, url, http.StatusTemporaryRedirect)
+		resp = serializer.NewTemporalRedirect(url)
```

Comment:
what is the point of using multiple layers of logic for it if you actually just do `http.Redirect` anyway?